testing/spectypes/premine-state.go

```diff
@@ -0,0 +1,426 @@
+	var body rooter
+	switch s.Version {
+	case version.Phase0:
+		body = &ethpb.BeaconBlockBody{
+			RandaoReveal: make([]byte, 96),
+			Eth1Data: &ethpb.Eth1Data{
+				DepositRoot: make([]byte, 32),
+				BlockHash:   make([]byte, 32),
+			},
+			Graffiti: make([]byte, 32),
+		}
+	case version.Altair:
+		body = &ethpb.BeaconBlockBodyAltair{
+			RandaoReveal: make([]byte, 96),
+			Eth1Data: &ethpb.Eth1Data{
+				DepositRoot: make([]byte, 32),
+				BlockHash:   make([]byte, 32),
+			},
+			Graffiti: make([]byte, 32),
+			SyncAggregate: &ethpb.SyncAggregate{
+				SyncCommitteeBits:      make([]byte, fieldparams.SyncCommitteeLength/8),
+				SyncCommitteeSignature: make([]byte, fieldparams.BLSSignatureLength),
+			},
+		}
+	case version.Bellatrix:
+		body = &ethpb.BeaconBlockBodyBellatrix{
+			RandaoReveal: make([]byte, 96),
+			Eth1Data: &ethpb.Eth1Data{
+				DepositRoot: make([]byte, 32),
+				BlockHash:   make([]byte, 32),
+			},
+			Graffiti: make([]byte, 32),
+			SyncAggregate: &ethpb.SyncAggregate{
+				SyncCommitteeBits:      make([]byte, fieldparams.SyncCommitteeLength/8),
+				SyncCommitteeSignature: make([]byte, fieldparams.BLSSignatureLength),
+			},
+			ExecutionPayload: &enginev1.ExecutionPayload{
+				ParentHash:    make([]byte, 32),
+				FeeRecipient:  make([]byte, 20),
+				StateRoot:     make([]byte, 32),
+				ReceiptsRoot:  make([]byte, 32),
+				LogsBloom:     make([]byte, 256),
+				PrevRandao:    make([]byte, 32),
+				BaseFeePerGas: make([]byte, 32),
+				BlockHash:     make([]byte, 32),
+				Transactions:  make([][]byte, 0),
+			},
+		}
+	default:
+		return errUnsupportedVersion
+	}
+
+	root, err := body.HashTreeRoot()
+	if err != nil {
+		return errors.Wrap(err, "could not hash tree root empty block body")
+	}
+	lbh := &ethpb.BeaconBlockHeader{
+		ParentRoot: params.BeaconConfig().ZeroHash[:],
+		StateRoot:  params.BeaconConfig().ZeroHash[:],
+		BodyRoot:   root[:],
+	}
+	return g.SetLatestBlockHeader(lbh)
+}
+
+func (s *premineGenesisConfig) setExecutionPayload(g state.BeaconState) error {
+	if s.Version < version.Bellatrix {
+		return nil
+	}
+	gb := s.GB
+
+	payload := &enginev1.ExecutionPayload{
+		ParentHash:    gb.ParentHash().Bytes(),
+		FeeRecipient:  gb.Coinbase().Bytes(),
+		StateRoot:     gb.Root().Bytes(),
+		ReceiptsRoot:  gb.ReceiptHash().Bytes(),
+		LogsBloom:     gb.Bloom().Bytes(),
+		PrevRandao:    params.BeaconConfig().ZeroHash[:],
+		BlockNumber:   gb.NumberU64(),
+		GasLimit:      gb.GasLimit(),
+		GasUsed:       gb.GasUsed(),
+		Timestamp:     gb.Time(),
+		ExtraData:     gb.Extra()[:32],
+		BaseFeePerGas: bytesutil.PadTo(bytesutil.ReverseByteOrder(gb.BaseFee().Bytes()), fieldparams.RootLength),
+		BlockHash:     gb.Hash().Bytes(),
+		Transactions:  make([][]byte, 0),
+	}
+	wep, err := blocks.WrappedExecutionPayload(payload)
+	if err != nil {
+		return err
+	}
+	eph, err := blocks.PayloadToHeader(wep)
+	if err != nil {
+		return err
+	}
+	wh, err := blocks.WrappedExecutionPayloadHeader(eph)
```

Comment:
The back and forth between wrapped and concrete types is rather awkward. In particular I wondered why `PayloadToHeader` gives the concrete type rather than `interfaces.ExecutionData`? Maybe obvious if I look at how it's used elsewhere but I haven't done that yet.